core/src/com/mygdx/game/models/Wall.java

```diff
@@ -0,0 +1,42 @@
+package com.mygdx.game.models;
+
+/**
+ * The Wall model represents a wall, and stores how damaged it currently is.
+ * maxHitpoints: The amount of hp the wall starts out with.
+ * hitPoints: The amount of hp the wall currently has.
+ * This class implements the MVC pattern.
+ */
+
+public class Wall {
+    private final int maxHitPoints;
+    private int hitPoints;
+
+    public Wall(int hitPoints) {
+        if(hitPoints <= 0)
+            throw new IllegalArgumentException("Hitpoints has to be initialized to a value above 0");
+
+        this.maxHitPoints = hitPoints;
+        this.hitPoints = hitPoints;
+    }
+
+    public int getMaxHitPoints() {
+        return maxHitPoints;
+    }
+
+    public int getHitPoints() {
+        return hitPoints;
+    }
+
+    // Returns true if the wall is broken by the damage, otherwise returns false
```

Comment:
```suggestion
    public boolean takeDmg(int dmg) {
    // Returns true if the wall is broken by the damage, otherwise returns false
        if(hitPoints <= 0)
            throw new IllegalArgumentException("Damage taken has to be above 0");
```

I think the comment should be inside the method, so its clear what it's referencing.

Reply:
I'm pretty sure the standard is to have documentation comments before what they are describing: https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html

I'll make the comment into a proper documentation comment though

Reply:
Perfect